server/src/main/java/org/opensearch/index/store/lockmanager/ShardLockInfo.java

```diff
@@ -0,0 +1,187 @@
+        this.fileToLock = fileName;
+    }
+
+    private void setResourceId(String resourceId) {
+        this.resourceId = resourceId;
+    }
+
+    private void setExpiryTime(String expiryTime) {
+        this.expiryTime = expiryTime;
+    }
+
+    private void setVersion(String version) {
+        this.version = version;
+    }
+
+    private Map<String, String> getLockData() {
+        Map<String, String> lockFileData = new HashMap<>();
+        lockFileData.put(RemoteStoreLockManagerUtils.RESOURCE_ID, resourceId);
+        lockFileData.put(RemoteStoreLockManagerUtils.FILE_TO_LOCK_NAME, fileToLock);
+        lockFileData.put(RemoteStoreLockManagerUtils.LOCK_EXPIRY_TIME, expiryTime);
+        lockFileData.put(RemoteStoreLockManagerUtils.VERSION,
+            version != null? version: RemoteStoreLockManagerUtils.DEFAULT_VERSION);
+        return lockFileData;
+    }
+
+    static Boolean isLockExpired(String lockName) {
+        Instant expiryTime = Instant.parse(LockFileUtils.getExpiryTimeFromLock(lockName));
+        return expiryTime.compareTo(Instant.now()) < 0;
+    }
+
+    @Override
+    public void writeLockContent(IndexOutput indexOutput) throws IOException {
+        indexOutput.writeMapOfStrings(getLockData());
+    }
+
+    static String getLockPrefixFromFileToLock(String fileToLock) {
+        return fileToLock + RemoteStoreLockManagerUtils.SEPARATOR;
+    }
+
+    static ShardLockInfo getLockFileInfoFromIndexInput(IndexInput indexInput) throws IOException {
+        Map<String, String> lockData = indexInput.readMapOfStrings();
+        ShardLockInfo shardLockInfo = new ShardLockInfo();
+        shardLockInfo.setfileToLock(lockData.get(RemoteStoreLockManagerUtils.FILE_TO_LOCK_NAME));
+        shardLockInfo.setResourceId(lockData.get(RemoteStoreLockManagerUtils.RESOURCE_ID));
+        shardLockInfo.setExpiryTime(lockData.get(RemoteStoreLockManagerUtils.LOCK_EXPIRY_TIME));
+        shardLockInfo.setVersion(lockData.get(RemoteStoreLockManagerUtils.VERSION));
+        return shardLockInfo;
+    }
+
+    static String getLockNameForResource(String[] lockFiles, String resourceId) {
+        Optional<String> originalLockName = Arrays.stream(lockFiles).filter(
+            lockFile -> resourceId.equals(LockFileUtils.getResourceIDFromLock(lockFile))).findAny();
+
+        if (originalLockName.isEmpty()) {
+            throw new ResourceNotFoundException("Lock For Resource is not present");
+        }
+        return originalLockName.get();
+    }
+
+    public static LockInfoBuilder getLockInfoBuilder() {
+        return new LockInfoBuilder();
+    }
+
+    @Override
+    public String getLockName() {
+        return LockFileUtils.generateLockName(fileToLock, resourceId, expiryTime);
+    }
+
+    static class LockFileUtils {
+        static String generateLockName(String mdFile, String resourceID, String expiryTime) {
+            return String.join(RemoteStoreLockManagerUtils.SEPARATOR, mdFile, resourceID, expiryTime)
```

Comment:
mainly as this will be used very frequently, as an optimisation, i am adding file_to_lock, resource_id and expiryTime in the lock Name. for example, to check if a metadata file is locked, we don't need to read the lock file every time to check its expiry. similarly, for releasing a lock for a resource, we can just list locks and filter on the basis of resourceIds.